user-impl/src/main/java/com/example/auction/user/impl/UserRepository.java

```diff
@@ -0,0 +1,192 @@
+        );
+    }
+
+    private static class PUserEventProcessor extends ReadSideProcessor<PUserEvent> {
+
+        private final CassandraSession session;
+        private final CassandraReadSide readSide;
+
+        private PreparedStatement insertUserStatement;
+
+        @Inject
+        public PUserEventProcessor(CassandraSession session, CassandraReadSide readSide) {
+            this.session = session;
+            this.readSide = readSide;
+        }
+
+        @Override
+        public ReadSideHandler<PUserEvent> buildHandler() {
+            return readSide.<PUserEvent>builder("pUserEventOffset")
+                    .setGlobalPrepare(this::createTables)
+                    .setPrepare(tag -> prepareStatements())
+                    .setEventHandler(PUserEvent.PUserCreated.class,
+                            e -> insertUser(e.getUser()))
+
+                    .build();
+        }
+
+        @Override
+        public PSequence<AggregateEventTag<PUserEvent>> aggregateTags() {
+            return PUserEvent.TAG.allTags();
+        }
+
+        private CompletionStage<Done> createTables() {
+            return doAll(
+                    session.executeCreateTable(
+                            "CREATE TABLE IF NOT EXISTS UserInfo (" +
+                                    "UserId UUID , " +
+                                    "createdAt Timestamp , " +
+                                    "Name text, " +
+                                    "email text " +
+                                    "PRIMARY KEY (createdAt, userId) " +
+                                    ")" +
+                            "WITH CLUSTERING ORDER BY (createdAt DESC)"
+                    )
+
+            );
+        }
+
+        private CompletionStage<Done> prepareStatements() {
+            return
+                    prepareInsertUserStatement();
+
+
+        }
+
+
+        private CompletionStage<Done> prepareInsertUserStatement() {
+
+            return session.
+                    prepare("INSERT INTO UserInfo(" +
+
+                            "UserId, " +
+                            "createdAt, " +
+                            "Name, " +
+                            "email" +
+
+                            ") VALUES (" +
+                            "?, " + // UserId
+                            "?, " + // createdAt
+                            "?, " + // Name
+                            "?" + // email
+
+                            ")"
+                    )
+                    .thenApply(accept(s -> insertUserStatement = s));
+        }
+
+
+        private CompletionStage<List<BoundStatement>> insertUser(PUser user) {
+            return completedStatements(
+                    Arrays.asList(insertUserCreator(user))
+
+            );
+        }
+
+        private BoundStatement insertUserCreator(PUser user) {
+            return insertUserStatement.bind(
+
+                    user.getId(),
+                    user.getCreatedAt(),
```

Comment:
In order to bind an `Instant` to a Cassandra query, you can use the optional [`InstantCodec`](http://docs.datastax.com/en/developer/java-driver/3.1/manual/custom_codecs/extras/#temporal-types) provided by the Java Cassandra driver.

You can see an example of how to register a codec in [`ItemRepository`](https://github.com/lagom/online-auction-java/blob/5db2bda15166ce9a632905ecf75aeed7f3cd5b07/item-impl/src/main/java/com/example/auction/item/impl/ItemRepository.java#L174) (and there is a similar example in `TransactionRepository`).